integration_test/third_party_apps_test.go

```diff
@@ -172,65 +172,61 @@ func installAgent(ctx context.Context, logger *logging.DirectoryLogger, vm *gce.
 	return nonRetryable, agents.InstallPackageFromGCS(ctx, logger, vm, agents.OpsAgentType, packagesInGCS)
 }
 
-// expectedLogs encodes a series of assertions about what data we expect
-// to see in the logging backend.
-type expectedLogs struct {
-	// Note on tags: the "yaml" tag specifies the name of this field in the
-	// .yaml file.
-	LogEntries []expectedLog `yaml:"log_entries"`
+type logFields struct {
+	Name        string `yaml:"name" validate:"required"`
+	Value       string `yaml:"value" validate:"required"`
```

Comment:
Since this is still treated as a regex, a name like `Matcher` or `ValueRegex` might be more appropriate.

Reply:
There's also a check in `constructQuery` for `Value != ""`, so you could actually make this optional as well. Or, for consistency with expected_metrics, we could default all `Value`s to `.*`. (The README says it's optional FWIW.)

Reply:
TL,DR: I will change Value to ValueRegex and make Value a required field

> [Optional. Not a very strong preference]
> 
> Making the field required could serve as a reminder for developers to verify the value in e2e whenever it is possible. It could potentially help boost test coverage. Developers can set it explicitly to `.*` when necessary.
> 
> This does mean the README needs to be changed to make it required though.
This is actually a very good point of view that i haven't thought of. I intended to make Value optional and description required but got it wrong..


> Since this is still treated as a regex, a name like `Matcher` or `ValueRegex` might be more appropriate.
`ValueRegex` SGTM

> There's also a check in `constructQuery` for `Value != ""`, so you could actually make this optional as well. Or, for consistency with expected_metrics, we could default all `Value`s to `.*`. (The README says it's optional FWIW.)
This is no build-in optional from https://pkg.go.dev/github.com/go-playground/validator/v10. For string, if the yaml file doesn't have it, it will default to "". If it's pointer, it will default to nil. Thinking that there will never be a case that logEntry field is intentionally set as "" and to be tested.

Reply:
How granular are the generated docs intended to be? I wonder if it would make sense to do per-leaf-field entries to solve the problem of objects, for example if something writes to `httpRequest` then have one entry for each field inside of `httpRequest`:

```yaml
- name: httpRequest.requestMethod
  value: GET|PUT|POST|...
- name: httpRequest.requestUrl
  value: .*
...
```

But that is a more complex question I think and we don't have to answer it here on the first cut.

I hadn't considered objects before so now I'm back to whether we should just make `ValueRegex` optional -- how about if the user doesn't specify it (or if they specify `""`), then instead of skipping the test, we check for field existence using e.g. `httpRequest:*`? This works for all kinds of fields, doesn't matter if it's object, string, int, etc.